collector/collector.go

```diff
@@ -48,22 +50,35 @@ type Collector interface {
 //
 //	gathers metrics from the source and writes to the writer.
 //	maxArchiveAge is the oldest time for which metrics are saved
-func NewCollector(source Source, writer Writer, maxArchiveAge time.Duration, log *logger.Logger) Collector {
+func NewCollector(sources []Source, writer Writer, maxArchiveAge time.Duration, log *logger.Logger) Collector {
 	return &collector{
-		Source:        source,
+		Sources:       sources,
 		Writer:        writer,
 		MaxArchiveAge: maxArchiveAge,
 		Logger:        log,
 	}
 }
 
 type collector struct {
-	Source
+	Sources []Source
 	Writer
 	MaxArchiveAge time.Duration
 	*logger.Logger
 }
 
+func mergeRelayCountsMaps(dayMaps []map[time.Time]map[string]api.RelayCounts) map[time.Time]map[string]api.RelayCounts {
+	mergedMap := make(map[time.Time]map[string]api.RelayCounts)
+	for _, dayMap := range dayMaps {
+		for day, appMap := range dayMap {
+			if _, ok := mergedMap[day]; !ok {
+				mergedMap[day] = make(map[string]api.RelayCounts)
+			}
+			maps.Copy(mergedMap[day], appMap)
```

Comment:
This needs to be fixed: an application could have relays listed on both InfluxDB (from v1) and TxDB (from v2). 
Please fix this and add unit tests to cover.

Reply:
Not a blocker for this version, but we need to:
1. merge the collector and api server into 1 binary with separate GO routines (and potentially improve their communication by using GO routine communication methods instead of DB table)

2. refactor: the api server only serves relay metrics. The collector should use its own mechanism of receiving relay reports from each TxDB region.